### PR TITLE
Automap: Differentiate stair types

### PR DESCRIPTION
**Pull request by glebm**
This is a re-implementation of @qndel's #1548 in the current codebase.

A current implementation should make it easier to implement whatever UI we decide on.

image

Refs #1362

### CODE REVIEW

File: Source/automap.cpp [hunk 1]
```diff
@@ -29,6 +29,8 @@ enum MapColors : uint8_t {
 	MapColorsPlayer = (PAL8_ORANGE + 1),
 	/** color for bright map lines (doors, stairs etc.) */
 	MapColorsBright = PAL8_YELLOW,
+	/** color fo the stairs type indicator line */
```
Comment (qndel) [inline Source/automap.cpp:32]:
```suggestion
	/** color of the stairs type indicator line */
```

File: Source/automap.cpp [hunk 2]
```diff
@@ -326,14 +328,95 @@ void DrawRiverForkOut(const Surface &out, Point center, uint8_t color)
...
+
+	// Hell: Parts of a staircase down
+	case 39: return StairsType::Down;
+	case 40: return StairsType::Part;
+	case 42: return StairsType::Down;
+	case 43:
+	case 44: return StairsType::Part;
+	case 46: return StairsType::Down;
+
+	// Parts of a staircase to town
```
Comment (qndel) [inline Source/automap.cpp:390]:
```suggestion
	// Hell: Parts of a staircase to town

```

**Comment by julealgon** [comment]
> until someone send a PR with better graphics?

Is this even _possible_, though? Wouldn't it make sense to first add some sort of support for sprite-based map elements or something like that?

Right now these "graphics" are 100% raw line draw calls right? It's not an image that you can set or anything like that.

On another note, what about adding a legend to the map? Maybe a way to say "blue == Upstairs, red == Downstairs", have it on by default, and allow it to be toggled off for people who are already used to the conventions. Just a thought...

**Comment by glebm** [comment]
> On another note, what about adding a legend to the map?

Yeah that's what I mean, let's punt any improvements to a future PR.

The idea is that having nothing at all is strictly worse than having some differentiation, even if confusing.

> Is this even possible, though? Wouldn't it make sense to first add some sort of support for sprite-based map elements or something like that?

It is entirely possible to draw a sprite there, simply calling `ClxDrawBlendedTRN` will render a sprite with 50% transparency (a version without `TRN` can be easily added). However, there is no support for zooming the sprite together with the map.

**Comment by glebm** [comment]
Retargeted to `development` and enabled auto-merge

**Comment by glebm** [comment]
Superseded by https://github.com/diasurgical/devilutionX/pull/6517

**Pull request closed without merging**